onyo/lib/commands.py

```diff
@@ -356,21 +352,24 @@ def get(inventory: Inventory,
     Raises
     ------
     """
+
+    # TODO: JSON output? Is this done somewhere?
     if sort_ascending and sort_descending:
+        # TODO: This should be solvable via a single switch instead. Plus:
```

Comment:
I am thinking of a `sort: Optional[bool] = None` for the python interface and no difference to the status quo with the CLI. :)

Reply:
The name `sort` would need some consideration, obv. But the point is, that's it's awkward to make the options mutually exclusive this way.

Reply:
Another option would be to make `sort` a choice between `'ascending', 'descending'` to not have the naming trouble.
But again: This is not about the CLI, but the python function. As far as the CLI goes, I'd like to have that validation (mutually exclusive) to be defined higher up, rather than "polluting" the code down here.
Somthing along these lines: https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.add_mutually_exclusive_group

(Didn't have a closer look at it yet, though)

Reply:
And then I read this: ;-P

> But again: This is not about the CLI, but the python function. 

Sooo.... about having argparse enforce this: I have no issue with that (or at least I can't remember a reason why we didn't).

Reply:
I think we are talking past each other (at least a bit). For me, it's primarily about the python interface. The awkwardness doesn't come from the fact, that we have mutually exclusive options per se, but that it's about two booleans enabling four combinations of which exactly two (thus, a single bool) are actually valid.
This can easily be done with an option, that simply doesn't allow to make an invalid call, rather than raising in half the cases. Sometimes it's necessary that way, but not here. It's about two mutually exclusive modes of sorting. This should be one option where you choose which one, because it's about the same thing.

This is independent from CLI. If you prefer the status quo for CLI, we can have `--sort-ascending`, but it would lead to a call in python like `onyo_get(sort='ascending')`.

However, I do like the `--sort type=ascending purchase_date=descending` suggestion, because it's a bit more powerful. Not sure whether that's ever needed, though.

Reply:
May be one more point WRT the python interface in case it's not obvious:
A consequence of the status quo is, that no proper defaults can be defined. If `sort_ascending` was `True` by default (as in `def onyo_get(sort_ascending: bool = True, ...)`), then one could never successfully pass `sort_descending=True` w/o also specifying `sort_ascending=False` explicitly. Which is awkward. Both options are currently required when the function is called, no matter what you want. If would specify defaults, then one could either call it w/o any of the two or with both changed to their respective opposite value.

Reply:
> one could never successfully pass sort_descending=True w/o also specifying sort_ascending=False explicitly. Which is awkward

I think this is a solid point.

I'm in favor of consolidating this for the Python API. Either what you suggested (`onyo_get(sort='ascending')`) or what I suggested (the Python equivalent of `--sort type=ascending purchase_date=descending`).

If it's convenient, it can be addressed in this PR. But I don't view it as a blocker, so if you'd rather create an issue (a la `natsort`), I'm fine with that as well.

Reply:
Ok. Then I'll go for `sort='ascending' | 'descending'` within this PR (because that's relatively easy), but make an issue about enhancing to the more powerful way of specifying it.